app/sources/DataInspector.m

```diff
@@ -249,6 +249,17 @@ static long double ieeeToLD(const void *bytes, unsigned exp, unsigned man) {
 
 static id floatingPointDescription(const unsigned char *bytes, NSUInteger length, enum Endianness_t endianness) {
     switch (length) {
+        case sizeof(short):
+        {
+            union {
+                uint16_t i;
+                __fp16 f;
+            } temp;
+            _Static_assert(sizeof temp.f == sizeof temp.i, "sizeof(uint16_t) is not 2!");
+            temp.i = *(const uint16_t *)bytes;
+            if (endianness != eNativeEndianness) temp.i = (uint16_t)reverse(temp.i, sizeof(uint16_t));
+            return [NSString stringWithFormat:@"%.15g", temp.f];
```

Comment:
I think this needs casting to `double`. In a test project if I do the same I get:

> Implicit conversion increases floating-point precision: '__fp16' to 'double'